gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,800 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Google Compute Engine operations. Operation identity can be obtained via {@link #operationId()}.
+ * For global operations {@link #operationId()} returns an {@link GlobalOperationId}, for region
+ * operations {@link #operationId()} returns a {@link RegionOperationId}, for zone operations
+ * {@link #operationId()} returns a {@link ZoneOperationId}. To get an {@code Operation} object with
+ * the most recent information use {@link #reload(Compute.OperationOption...)}.
+ */
+public final class Operation implements Serializable {
+
+  private static final long serialVersionUID = -8979001444590023899L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private transient Compute compute;
+  private final ComputeOptions options;
+  private final String id;
+  private final OperationId operationId;
+  private final Long creationTimestamp;
+  private final String clientOperationId;
+  private final String operationType;
+  private final String targetLink;
+  private final String targetId;
+  private final Status status;
+  private final String statusMessage;
+  private final String user;
+  private final Integer progress;
+  private final Long insertTime;
+  private final Long startTime;
+  private final Long endTime;
+  private final List<OperationError> errors;
+  private final List<OperationWarning> warnings;
+  private final Integer httpErrorStatusCode;
+  private final String httpErrorMessage;
+  private final String description;
+
+  /**
+   * Types of operations.
```

Comment:
"Types of operation statuses"?  Or just "Operation statuses"? ("Types of operations" sounds closer to global vs zone vs region)